src/components/vr-scatter-plot/vr-scatter-plot.component.ts

```diff
@@ -1,32 +1,88 @@
+import { Component, Input, OnChanges, OnDestroy, OnInit, SimpleChanges, ElementRef, ViewChild } from '@angular/core';
 import { Scatterplot } from '../../d3/scatterplot.d3';
-import 'aframe';
-
-
+import { PreferenceService } from '../../services/preference/preference.service';
+import { DataService } from '../../services/data/data.service';
+import { Meta } from '../../datasets/metas/types';
+import { TimeSeriesQueryOptions } from '../../datasets/queries/time-series.query';
+import { LineChartDatum} from '../line-chart/line-chart.component';
+import { LineChartMeta } from '../../datasets/metas/line-chart.meta';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { DAY } from '../../utils/timeUnits';
+import { map, takeUntil } from 'rxjs/operators';
+import { DATA_PREFERENCE } from '../../i18n';
+import { datasets } from '../../datasets';
 
 @Component({
   selector: 'app-vr-scatter-plot',
-  templateUrl: './vr-scatter-plot.component.html'
+  templateUrl: './vr-scatter-plot.component.html',
+  styleUrls: ['./vr-scatter-plot.component.scss'],
 })
 export class VRScatterPlotComponent implements OnInit, OnChanges, OnDestroy{
-  private vrScatterPlot: Scatterplot;
-  private shape: string;
-  private color: string;
+  @Input() endDate = new Date();
+  @Input() startDate = new Date(this.endDate.getTime() - 30 * DAY);
+  @Input() datasetPref: LineChartMeta;
+  dataset$ = this.preferenceService.dataset$;
+  DATA_PREFERENCE = DATA_PREFERENCE;
+  vrScatterPlot: Scatterplot;
+  componentMetas: Meta[];
+  queryOptions$ = new BehaviorSubject<TimeSeriesQueryOptions>({
+    range: [this.startDate, this.endDate],
+  });
+  datum$ = new BehaviorSubject<LineChartDatum>({
+    label: '',
+    points: [],
+  });
+  private destroy$ = new Subject();
+  dataService: DataService;
+  @ViewChild('ascene') ascene: ElementRef;
 
   constructor(
+    private preferenceService: PreferenceService
   ) {
+    this.preferenceService = preferenceService;
+    this.dataService = new DataService(this.preferenceService);
     this.vrScatterPlot = new Scatterplot('a-sphere');
   }
+
   ngOnInit() {
-   this.vrScatterPlot.init(document.querySelector('a-scene'), [10, 20, 30, 40, 50, 60]);
+    this.dataService.dataset$
+    .pipe(takeUntil(this.destroy$))
+    .subscribe(dataset => {
+      // componentMetas is initialized to different dataset metas - will help funnel dataset
+      this.componentMetas = dataset.metas;
+      // dataset.metas[0].type = 'tabbed' and dataset.metas[1] = 'line' if chosen UserWhiteNoise
+      // dataset.metas[0] - 'line' if Dummy chosen
+      if (dataset.metas[0].type === 'tabbed') {
+        this.datasetPref = (dataset.metas[0] as any).metas[0];
+      } else if (dataset.metas[0].type === 'line') {
+        this.datasetPref = dataset.metas[0] as LineChartMeta;
+      }
+      // calling this.init() from inside bc when code above is in constructor,
```

Comment:
Can probably remove this comment, and rename `init` to something more specific.

Reply:
went ahead and removed comment and renamed `.init()` to `.setDataOnD3()`